### ISSUE TITLE
any way to pass fields through CLI or file?

### ISSUE THREAD

**Issue by vesper8** [mitulgolakiya/laravel-api-generator#26]
I really like your generator, but right now I am doing a lot of trial and error, figuring out how to make the most of your package by testing and regenerating.. testing the views and crud.. making tweaks to the generator and then re-running it. And it seems the only way to pass the fields and field options is one at a time via the interactive mode which is really cumbersome if using a lot of fields.

I would like to be able to pass it all on one line.. or to point the generator to a file containing the same parameters as what the inractive mode asks for. It could be in the exact same format.. as long as I don't have to re-type it manually every time. Is this already possible?

On a side note.. there's no way you could possibly generate api/scaffold by being passed an existing table and then iterating over the fields? Well at least if you could fulfill my first request.. it would be fairly easy to generate parameters from an existing schema.

Thanks for your work!!

**Comment by mitulgolakiya** [comment]
@vesper8,
unfortunately, currently passing fields are only possible via interactive console and there is no way to make models from existing tables.

But I will take it in next roadmap.

**Comment by vesper8** [comment]
I was just looking at https://github.com/Indatus/blacksmith which does something similar to your project. That package also allows to pass all the field options on the command line. Anyway this is not very hard to implement.. I am doing a hack implementation of it right now.. but it's not going to be pretty because I'm in a rush.. still it's not hard to take a filename as input as load field options from that file.

In any case I tried Blacksmith and it's good.. but I prefer your implementation and of course I like that your project has active development.. anyway you can maybe be inspired by Blacksmith. I like that Blacksmith can auto-generate tests too.

Well keep it up! I look forward to your next release

**Comment by vesper8** [comment]
I did this very quickly and I'm sure you can make it much better but maybe this will help you a bit or motivate you to make a new release :)

I simply added a 2nd command line optional parameter called modelDefinition. And if it is not empty I am iterating through each line instead of using interactive mode.

I am ignoring validations for now.

```
    public function getInputFields_custom()
    {
        $fields = [];

        $this->commandObj->info("Loading fields from {$this->modelDefinition}");

        if ($handle = fopen( $this->modelDefinition, "r")) {

            while (($line = fgets($handle)) !== false)
            {
                $fieldInputStr = str_replace(array("\r", "\n"), "", $line);

                $this->commandObj->info("Processing... {$fieldInputStr}");

                if(empty($fieldInputStr))
                    break;

                $fieldInputs = explode(":", $fieldInputStr);

                if(sizeof($fieldInputs) < 2)
                {
                    $this->commandObj->error("Invalid Input. Try again");
                    continue;
                }

                $fieldName = $fieldInputs[0];

                $fieldTypeOptions = explode(",", $fieldInputs[1]);
                $fieldType = $fieldTypeOptions[0];
                $fieldTypeParams = [];
                if(sizeof($fieldTypeOptions) > 1)
                {
                    for($i = 1; $i < sizeof($fieldTypeOptions); $i++)
                        $fieldTypeParams[] = $fieldTypeOptions[$i];
                }

                $fieldOptions = [];
                if(sizeof($fieldInputs) > 2)
                    $fieldOptions[] = $fieldInputs[2];

                $validations = "";

                $field = [
                    'fieldName'       => $fieldName,
                    'fieldType'       => $fieldType,
                    'fieldTypeParams' => $fieldTypeParams,
                    'fieldOptions'    => $fieldOptions,
                    'validations'     => $validations
                ];

                $fields[] = $field;
            }

            fclose($handle);

        } else {
                // error opening the file.
                $this->commandObj->info("Problem opening {$this->modelDefinition}");
        }


        return $fields;
    }
```

**Comment by aitorjs** [comment]
I make some modifications on laravel-api-generator package to have fields for textarea, select, date, check... https://github.com/hazicms/generator. I'm on active development and I need to fix two small errors to can publish v1.0 but is the package is usefull.

Still you need to type the field type but is on the way for what you need. I have field configuration file on mind.

**Comment by vesper8** [comment]
That's awesome! I'll be looking forward to both your solutions!

**Comment by vesper8** [comment]
was this added in the latest release? :) or coming soon on the roadmap?

**Comment by mitulgolakiya** [comment]
It's in next release roadmap... Will release it soon :)

**Comment by mitulgolakiya** [comment]
it's complete in 1.3 branch and will be added in next release. If you want early access, add dev branch to your composer,

`"mitulgolakiya/laravel-api-generator": "1.3.*@dev"`

I have also added documentation [here](https://github.com/mitulgolakiya/laravel-api-generator#passing-fields-from-file).

Check and let me know.

**Comment by mitulgolakiya** [comment]
Released in 1.3.2.
